### DISCUSSION TITLE
Pyabel vs. analytical integral comparisons

### DISCUSSION THREAD

**Discussion by evavagiakis** [deepskies/DeepSZSim#206] [Ideas]
To evaluate the differences between the submaps generated using pyabel versus our homemade/better understood (but much slower) integral, we could: 

- [x] Generate a set of sims for a range of (z, M) using each approach, all other factors equal (ignore CMB, noise etc.).
- [x] Compare the central pixel value (y0) between these approaches, plotting the difference as a function of z and then as a function of M. 
- [x] Repeat with beam convolution
- [x] Apply the aperture photometry filter (https://github.com/deepskies/DeepSZSim/blob/main/deepszsim/filters.py) to the submaps. Ideally, this filter aperture would be selected based on the redshift/angular diameter distance of the sample. To avoid needing to do that, we could plot the difference in returned value (dT) as a function of z and then as a function of M for a selection of apertures; say, 2, 3 and 4 arcmin.

**Comment by samueldmcdermott** [comment]
@evavagiakis I've completed all these steps -- as anticipated, the differences between our results and `pyabel` become smaller with convolution and with aperture photometry applied, though they're still nonzero (and the difference depends on mass and redshift). Would you like to see all of these plots in this reply as individual plots, or can I send an html of my ipynb, or show them to you on a call sometime?

One thing worth highlighting: I found that `pyabel` has the nonphysical behavior that the y value is _lower_ when we integrate along a longer los, so that's a pretty bright red flag (presumably attributable to some poor choice of basis functions for the transform that don't approximate our pressure profiles well). At this point I'll probably cease investigating `pyabel` and just move on to using our integration method (unless there's something you're interested to see)
